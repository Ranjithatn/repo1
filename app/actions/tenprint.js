// @flow
import {
  REQUEST_TENPRINT_VERIFY,
  REQUEST_SET_TENPRINT_VERIFY,
  REQUEST_SET_TENPRINT_VERIFY_HAND,
  REQUEST_SET_TENPRINT_VERIFY_STORE,
  REQUEST_SET_TENPRINT_VERIFY_MATCH,
  REQUEST_SET_TENPRINT__VERIFY_MATCHED_DATA,
  REQUEST_REMOVE_TENPRINT__VERIFY_MATCHED_DATA,
  REQUEST_SHOW_ADJUDICATOR,
  REQUEST_LOAD_LIVESCAN,
  REQUEST_MATCHED_PERSON_CHANGED,
  RECIVED_MATCHED_PERSON_CHANGED,
  REQUEST_MODAL_DATA,
  RECIVED_MODAL_DATA,
  REQUEST_HIDE_ADJUDICATOR,
  RECIEVE_NOTE_STORE,
  REQUEST_SHOW_ADJUDICATOR_SUCCESS,
  REQUEST_SET_ADD_NEW,
  REQUEST_DELETE_NEW_CRIME_DATA,
  REQUEST_SELECTED_TENPRINT_TYPE,
  REQUEST_SET_TENPRINT_VERIFY_RAW,
  MODAL_CHECKBOX,
  TOGGLE_PANE_VISIBILITY,
  REQUEST_RESET_CRIME_DATA,
  REQUEST_STORE_LINK_DATA,
  REQUEST_RESET_LINK_DATA,
  REQUEST_UNMATCH
  
} from "./actionTypes";

export const requestTenprintVerify = historyDetails => ({
  type: REQUEST_TENPRINT_VERIFY,
  payload: historyDetails
});
export const requestShowAdjudicator = adjudicatorData => ({
  type: REQUEST_SHOW_ADJUDICATOR,
  payload: adjudicatorData
});
export const requestShowAdjudicatorSuccess = adjudicatorSaveData => ({
  type: REQUEST_SHOW_ADJUDICATOR_SUCCESS,
  payload: adjudicatorSaveData
});

export const requestSetTenprintVerify = tenprintData => ({
  type: REQUEST_SET_TENPRINT_VERIFY,
  payload: tenprintData
});

export const requestSetTenprintVerifyHand = TenprintHandData => ({
  type: REQUEST_SET_TENPRINT_VERIFY_HAND,
  payload: TenprintHandData
});

export const requestSetTenprintVerifyStoreHand = TenprintVerifyStoreData => ({
  type: REQUEST_SET_TENPRINT_VERIFY_STORE,
  payload: TenprintVerifyStoreData
});

export const requestSetTenprintVerifyMatchData = TenprintVerifyMatchData => ({
  type: REQUEST_SET_TENPRINT_VERIFY_MATCH,
  payload: TenprintVerifyMatchData
});

export const requestSetTenprintVerifyStoreData = TenprintVerifyStoreData => ({
  type: REQUEST_SET_TENPRINT__VERIFY_MATCHED_DATA,
  payload: TenprintVerifyStoreData
});

export const requestRemoveTenprintVerifyMatch = () => ({
  type: REQUEST_REMOVE_TENPRINT__VERIFY_MATCHED_DATA
});

//tenprint verify new
export const requestSelectedTenprintType = selected => ({
  type: REQUEST_SELECTED_TENPRINT_TYPE,
  payload: selected
});
export const requestModalCheckbox = selected => ({
  type: MODAL_CHECKBOX,
  payload: selected
});

export const requestMatchedPersonChanged = (
  index,
  PersonDetails,
  event,
  PersonID
) => {
 
  const data = {
    Index:index,
    PersonInfo:PersonDetails,
    Selected:event.target.checked,
    userID:PersonID
  };
  return {
    type: REQUEST_MATCHED_PERSON_CHANGED,
    payload: data
  };
};

export const recivedMatchedPersonChanged = (data) => {
  return {
    type: RECIVED_MATCHED_PERSON_CHANGED,
    payload: data
  };
};
export const requestModalData = () => {
  return {
    type: REQUEST_MODAL_DATA
  };
};
export const recivedModalData = (data) => {
  return {
    type: RECIVED_MODAL_DATA,
    payload: data
  };
};
export const requestHideAdjudicator = status => {
  return {
    type: REQUEST_HIDE_ADJUDICATOR,
    payload: status
  };
};
export const requestStoreNote = NoteData => {
  return {
    type: RECIEVE_NOTE_STORE,
    payload:NoteData
  };
};

export const requestSetTenprintVerifyRaw = data => ({
  type: REQUEST_SET_TENPRINT_VERIFY_RAW,
  payload: data
});
export const requestSetAddnew = () => ({
  type: REQUEST_SET_ADD_NEW
});


export const requestTogglePaneVisibility = data => ({
  type: TOGGLE_PANE_VISIBILITY,
  payload: data
});
export const requestDeleteNewCrimeData = index => ({
  type: REQUEST_DELETE_NEW_CRIME_DATA,
  payload: index
});
export const requestResetCrimeType = () => ({
  type: REQUEST_RESET_CRIME_DATA
});
export const requestResetlinkData = () => ({
  type: REQUEST_RESET_LINK_DATA
});
export const requestUnMatch = () => ({
  type: REQUEST_UNMATCH
});


export const requestStoreLinkData = (
 latentId,bioId
) => {
 
  const data = {
    latentId:latentId,
    bioId:bioId
  };
  return {
    type: REQUEST_STORE_LINK_DATA,
    payload: data
  };
};