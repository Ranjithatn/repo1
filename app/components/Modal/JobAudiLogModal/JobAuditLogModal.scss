.newAction-modal footer.modal-card-foot {
    display: flex;
    justify-content: flex-end;
  }