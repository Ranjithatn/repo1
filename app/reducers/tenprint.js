// @flow
import {
  REQUEST_SET_TENPRINT_VERIFY,
  REQUEST_SET_TENPRINT_VERIFY_STORE,
  REQUEST_SET_TENPRINT__VERIFY_MATCHED_DATA,
  REQUEST_REMOVE_TENPRINT__VERIFY_MATCHED_DATA,
  REQUEST_SHOW_ADJUDICATOR_SUCCESS,
  RECIVED_MATCHED_PERSON_CHANGED,
  RECIVED_MODAL_DATA,
  REQUEST_HIDE_ADJUDICATOR,
  RECIEVE_NOTE_STORE,
  //tenprint verify new
  REQUEST_SELECTED_TENPRINT_TYPE,
  REQUEST_SET_TENPRINT_VERIFY_RAW,
  MODAL_CHECKBOX,
  REQUEST_RESET_LINK_DATA,
  TOGGLE_PANE_VISIBILITY
} from "../actions/actionTypes";
const initialState = {
  tenprintverifydata: {},
  tenprintverifyMatchInfo: {
    Results: []
  },
  tenprintverifyHandData: {
    position: "Left"
  },
  ShowAdjudicator: false,
  SelectedRow: 0,
  AdjudicatorPassImageData: {
    Result: {
      matched: {
        image: ""
      }
    }
  },
  MatchedUserRowID: "",
  MatchedPersonInfo: {},
  adjudicatorMatch: [],
  ConfirmModalData: {},
  selectedSearchOption: undefined,
  //tenprint verify new
  showTenprintSelected: {
    selected: "fingerprint",
    title: "titlebio.fingerprint"
  },
  adjudicatorResult: {},

  //update and register modal lashing, fine initial state
  ModalCheckbox: {
    lashing: false,
    fine: false
  },

  panes: {
    sidebar: true,
    table: true
  }
};

export default function latent(state = initialState, action = {}) {
  switch (action.type) {
    case REQUEST_SET_TENPRINT_VERIFY_RAW:
      return {
        ...state,
        tenprint: action.payload
      };

    case REQUEST_SET_TENPRINT_VERIFY:
 
      return {
        ...initialState,
        tenprintverifydata: action.payload,
        adjudicatorMatch: action.payload.leftmatches.fingers
      };
    case REQUEST_SET_TENPRINT_VERIFY_STORE:
      return toggleHannds(state, action.payload);
    case REQUEST_SET_TENPRINT__VERIFY_MATCHED_DATA:
      return {
        ...state,
        tenprintverifyMatchInfo: action.payload,
        SelectedRow: action.payload.SelectedRow,
        ShowAdjudicator: false
        // adjudicatorMatch: action.payload.Results
      };
    case REQUEST_REMOVE_TENPRINT__VERIFY_MATCHED_DATA:
      return removeMatches(state);
    case REQUEST_SHOW_ADJUDICATOR_SUCCESS:
      return showAdjudicator(state, action.payload);
    case REQUEST_HIDE_ADJUDICATOR:
      return {
        ...state,
        ShowAdjudicator: action.payload
      };

    case RECIVED_MODAL_DATA:
      return { ...state, ConfirmModalData: action.payload };

    case RECIEVE_NOTE_STORE:
      return { ...state, ActionNote: action.payload };

    case RECIVED_MATCHED_PERSON_CHANGED:
      return {
        ...state,
        MatchedUserRowID: action.payload.Index,
        MatchedPersonInfo: action.payload.PersonInfo,
        MatchedPersonUserID: action.payload.userID,
      };
    case TOGGLE_PANE_VISIBILITY:
      let panes = { ...state.panes };
      if (action.payload.pane === "all") {
        panes.sidebar = action.payload.visible;
        panes.table = action.payload.visible;
      } else {
        panes[action.payload.pane] = action.payload.visible;
      }

      return { ...state, panes };
    default:
      return state;
    //tenprint verify new
    case REQUEST_SELECTED_TENPRINT_TYPE:
      return { ...state, showTenprintSelected: action.payload };
    case MODAL_CHECKBOX:
      const { type, status } = action.payload;
      let newState = Object.assign({}, state);
      newState.ModalCheckbox[type] = status;
      return newState;
  }
}
const removeMatches = state => {
  delete state.tenprintverifyMatchInfo;
  return {
    ...state
  };
};
const showAdjudicator = (state, payload) => {
  state.ShowAdjudicator = payload.status;
  state.AdjudicatorPassImageData = payload;
  state.adjudicatorMatch = payload.Result.matched;
  state.adjudicatorProbe = payload.Result.probe;
  state.adjudicatorResult = payload.Result;
  return { ...state };
};
const toggleHannds = (state, payload) => {
  state.tenprintverifyHandData = payload;
  state.ShowAdjudicator = false;
  state.AdjudicatorPassImageData.status = false;
  return { ...state };
};
