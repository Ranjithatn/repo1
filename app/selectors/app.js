import { createSelector } from "reselect";

export const serverStatusSelector = createSelector(
	state => state.app.serverStatus,
	serverStatus => serverStatus
);


export const nfiqQualityThresholdSelector = createSelector(
	state => state.app.settings.nfiq_quality_threshold,
	nfiq_quality_threshold => nfiq_quality_threshold
);
