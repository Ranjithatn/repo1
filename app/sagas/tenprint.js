//

import { call, put, takeLatest, select, takeEvery } from "redux-saga/effects";
import { push } from "react-router-redux";
import * as localStorage from "../utils/localStorage";
import { delay } from "redux-saga";
import { requestShowNotification } from "../actions/notifications";
import Api, { url } from "../api/";
import { startSpinner, stopSpinner } from "../../app/actions/spinner";
import { requestShowModal } from "../actions/modal";
import { requestTogglePaneVisibility } from "../actions/tenprint";


import {
  setPersonData,
  setCustomSearchDatabase,
  requestSetCustomSearchID,
  setCustomSearchResponse,
} from "../actions/customSearch";
import {
  REQUEST_TENPRINT_VERIFY,
  REQUEST_SET_TENPRINT_VERIFY,
  REQUEST_SET_TENPRINT_VERIFY_HAND,
  REQUEST_SET_TENPRINT_VERIFY_STORE,
  REQUEST_SET_TENPRINT_VERIFY_MATCH,
  REQUEST_MATCHED_PERSON_CHANGED,
  REQUEST_MODAL_DATA,
  REQUEST_SHOW_ADJUDICATOR,
  REQUEST_UNMATCH
} from "../actions/actionTypes";

import {
  requestTenprintVerify,
  requestSetTenprintVerify,
  requestSetTenprintVerifyHand,
  requestSetTenprintVerifyStoreHand,
  requestSetTenprintVerifyStoreData,
  requestRemoveTenprintVerifyMatch,
  recivedMatchedPersonChanged,
  recivedModalData,
  requestStoreNote,
  requestShowAdjudicatorSuccess,
  requestSetTenprintVerifyRaw
} from "../actions/tenprint";
import { resetActionType } from "../actions/jobs";

import {
  selectedJobStatusSelector,
  selectedActionSelector,
  jobsTypeSelector,
  ActionSelector,
  selectedJobSelector
} from "../selectors/jobs";
import {
  MatchedPersonInfoSelector,
  SelectedRowSelector
} from "../selectors/tenprint";
import { translate, translateRes } from "../utils/intl";
import { setTimeout } from "timers";

export function* watchrequestUnMatch() {
  yield takeLatest(REQUEST_UNMATCH, callrequestUnMatch);
}
export function* watchRequestTenprintVerify() {
  yield takeLatest(REQUEST_TENPRINT_VERIFY, callrequestTenprintVerify);
}

export function* watchRequestTenprintVerifyHand() {
  yield takeLatest(
    REQUEST_SET_TENPRINT_VERIFY_HAND,
    callrequestTenprintVerifyHand
  );
}

export function* watchRequestTenprintVerifyMatchData() {
  yield takeLatest(
    REQUEST_SET_TENPRINT_VERIFY_MATCH,
    callrequestTenprintVerifyMatchData
  );
}

export function* watchRequestMatchedPersonChanged() {
  yield takeLatest(
    REQUEST_MATCHED_PERSON_CHANGED,
    callRequestMatchedPersonChanged
  );
}

export function* watchRequestModalData() {
  yield takeLatest(REQUEST_MODAL_DATA, callRequestModalData);
}

export function* watchRequestShowAdjudicator() {
  yield takeLatest(REQUEST_SHOW_ADJUDICATOR, callRequestShowAdjudicator);
}

export function* callrequestTenprintVerifyHand(action) {
  try {
    const { payload: verifyhand } = action;
    yield put(requestSetTenprintVerifyStoreHand(verifyhand));
    yield put(requestRemoveTenprintVerifyMatch());
  } catch (e) {
    console.log(e);
  }
}

export function* callrequestTenprintVerifyMatchData(action) {
  try {
    const { payload: matchdata } = action;
    yield put(requestSetTenprintVerifyStoreData(matchdata));
  } catch (e) {
    console.log(e);
  }
}

export function* callrequestTenprintVerify(action) {
  yield put(startSpinner());

  if (
    action.payload.status === "New" ||
    action.payload.status === "In Progress"
  ) {
    yield put(
      requestShowNotification({
        message: action.payload.formatMessage({ id: "preocessingData" }),
        type: "is-warning"
      })
    );
    yield put(stopSpinner());
    return;
  }
  if (action.payload.status === "Failed") {
    yield put(requestStoreNote(action.payload.note));
    yield put(requestShowModal({ modalType: "NOTE_SHOW_MODAL" }));

    yield put(stopSpinner());
    yield put(stopSpinner());
    return;
  }
  let jobType = yield select(jobsTypeSelector);
  if (jobType !== "Custom") {
    let TenprintData = {};
    let JobStatus = yield select(selectedJobStatusSelector);
    let ActionTpe = yield select(ActionSelector);

    if (
      ActionTpe !== "Search Criminal" &&
      ActionTpe !== "Search Latent" &&
      ActionTpe !== "Search Civil"
    ) {
      const { payload: historydata } = action;

      const res = yield call(Api, {
        url: url.action.actionResult,
        method: "POST",
        isJwtRequired: true,
        data: historydata.JOBID
      });
      if (res) {
        if (res.note !== undefined && res.note !== "" && res.note !== null) {
          yield put(requestStoreNote(res.note));
          yield put(stopSpinner());
          yield put(requestShowModal({ modalType: "NOTE_SHOW_MODAL" }));
        } else {
          yield put(stopSpinner());
        }
      }
    } else {
      let JobType = yield select(jobsTypeSelector);
   
      if (JobStatus !== "New" && JobStatus !== undefined) {
        const { payload: historydata } = action;

        let LeftFingerInfo = {
          fingers: []
        };
        let RightFingerInfo = {
          fingers: []
        };
        let Leftmatches = {
          fingers: []
        };
        let Rightmatches = {
          fingers: []
        };
        let Personinfos = {};
        let matchedData = {};
        let matchedPosition = "";

        try {
          const res = yield call(Api, {
            url: url.action.actionResult,
            method: "POST",
            isJwtRequired: true,
            data: historydata.JOBID
          });
          let impressionType;

          if (res.probe && res.probe.biometrics.length > 14) {
            impressionType = "Roll";
          } else {
            impressionType = "Flat";
          }

          res.probe &&
            res.probe.biometrics.forEach((PresentObject, index) => {
              if (
                res.fingerprintMatches &&
                res.fingerprintMatches.length !== 0
              ) {
               
                if (
                  res.fingerprintMatches[index] &&
                  res.fingerprintMatches[index].matchInfos[0] !== "" &&
                  res.fingerprintMatches[index].matchInfos[0] !== undefined
                ) {
                  matchedPosition =
                    res.fingerprintMatches[index].matchInfos[0]
                      .fingerprintPosition;
                  matchedData = res.fingerprintMatches[index].matchInfos;

                  matchFingerStore(matchedPosition, matchedData);
                }
              }
              let temp_data = PresentObject.position;
              temp_data = temp_data.replace(/\s/g, "");
              switch (temp_data) {
                case "LeftThumb":
                  // LeftFingerInfo.Thumb.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    LeftFingerInfo.fingers.push(PresentObject);
                  }

                  break;
                case "LeftIndex":
                  // LeftFingerInfo.Index.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    LeftFingerInfo.fingers.push(PresentObject);
                  }

                  // Leftmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "LeftMiddle":
                  // LeftFingerInfo.Middle.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    LeftFingerInfo.fingers.push(PresentObject);
                  }

                  // Leftmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "LeftRing":
                  // LeftFingerInfo.Ring.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    LeftFingerInfo.fingers.push(PresentObject);
                  }

                  // Leftmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "LeftLittle":
                  // LeftFingerInfo.Small.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    LeftFingerInfo.fingers.push(PresentObject);
                  }

                  // Leftmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "RightThumb":
                  // RightFingerInfo.Thumb.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    RightFingerInfo.fingers.push(PresentObject);
                  }

                  // Rightmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "RightIndex":
                  // RightFingerInfo.Index.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    RightFingerInfo.fingers.push(PresentObject);
                  }

                  // Rightmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "RightMiddle":
                  // RightFingerInfo.Middle.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    RightFingerInfo.fingers.push(PresentObject);
                  }

                  // Rightmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "RightRing":
                  // RightFingerInfo.Ring.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    RightFingerInfo.fingers.push(PresentObject);
                  }

                  // Rightmatches.fingers.push(PresentObject.matchInfos);
                  break;

                case "RightLittle":
                  // RightFingerInfo.Small.push(PresentObject)
                  if (PresentObject.impression === impressionType) {
                    RightFingerInfo.fingers.push(PresentObject);
                  }

                  // Rightmatches.fingers.push(PresentObject.matchInfos);
                  break;
                case "UnknownFinger":
                  LeftFingerInfo.fingers.push(PresentObject);

                  break;
              }

              let latentProbeData = [];

              if (
                res.probe.biometrics.length === 1 &&
                LeftFingerInfo.fingers.length === 0 &&
                RightFingerInfo.fingers.length === 0
              ) {
                LeftFingerInfo.fingers.push(res.probe.biometrics[0]);
              }

              if (
                res.probe.biometrics.length === 1 &&
                LeftFingerInfo.fingers.length === 0
              ) {
                latentProbeData = res.probe.biometrics;
              }

              TenprintData = {
                leftfingers: LeftFingerInfo,
                rightfingers: RightFingerInfo,
                leftmatches: Leftmatches,
                rightmatches: Rightmatches,
                personinfos: res.matchPersons,
                ProbeMugshotInfo: res.probe.mugshot,
                ProbePersonInfo: res.probe.personInfo,
                database: res.database,
                latentProbeData: latentProbeData,
                matched:
                  (res.fingerprintMatches[index] &&
                    res.fingerprintMatches[index].matchInfos[0]) ||
                  ""
              };
              
              // yield put(requestSetTenprintVerify(TenprintData))
            });

          function matchFingerStore(temp_data, object) {
           
            switch (temp_data) {
              case "LeftThumb":
                Leftmatches.fingers.push(object);
                break;
              case "Left Thumb":
                Leftmatches.fingers.push(object);
                break;
              case "LeftIndex":
                Leftmatches.fingers.push(object);
                break;
              case "Left Index":
                Leftmatches.fingers.push(object);
                break;
              case "LeftMiddle":
                Leftmatches.fingers.push(object);
                break;
              case "Left Middle":
                Leftmatches.fingers.push(object);
                break;
              case "LeftRing":
                Leftmatches.fingers.push(object);
                break;
              case "Left Ring":
                Leftmatches.fingers.push(object);
                break;
              case "LeftLittle":
                Leftmatches.fingers.push(object);
                break;
              case "Left Little":
                Leftmatches.fingers.push(object);
                break;
              case "RightThumb":
                Rightmatches.fingers.push(object);
                break;
              case "Right Thumb":
                Rightmatches.fingers.push(object);
                break;
              case "RightIndex":
                Rightmatches.fingers.push(object);
                break;
              case "Right Index":
                Rightmatches.fingers.push(object);
                break;
              case "RightMiddle":
                Rightmatches.fingers.push(object);
                break;
              case "Right Middle":
                Rightmatches.fingers.push(object);
                break;
              case "RightRing":
                Rightmatches.fingers.push(object);
                break;
              case "Right Ring":
                Rightmatches.fingers.push(object);
                break;
              case "RightLittle":
                Rightmatches.fingers.push(object);
                break;
              case "Right Little":
                Rightmatches.fingers.push(object);
                break;
              case "Left Four Fingers":
                Leftmatches.fingers.push(object);
                break;
              case "Right Four Fingers":
                Leftmatches.fingers.push(object);
                break;
              case "Both Thumbs":
                Leftmatches.fingers.push(object);
                break;
              case "UnknownFinger":
                Leftmatches.fingers.push(object);
                Rightmatches.fingers.push(object);
                break;
              case "Unknown Finger":
                Leftmatches.fingers.push(object);
                Rightmatches.fingers.push(object);
                break;
            }
          }

          
          if (
            res.probe &&
            res.probe.biometrics &&
            res.probe.biometrics.length !== 0
          ) {
            yield put(stopSpinner());
            
            yield put(requestSetTenprintVerify(TenprintData));
         
            yield put(requestSetTenprintVerifyRaw(res));
            yield put(stopSpinner());
           
            if (JobType === "Tenprint") {
              yield put(push("/authenticated/tenprint/tenprintverify"));
            } else if (JobType === "Latent") {
              yield put(push("/authenticated/latent/latentverify"));
            }
          } else {
            yield put(stopSpinner());
            yield put(
              requestShowNotification({
                message: "FingerPrint matches not recived",
                type: "is-warning"
              })
            );
          }
        } catch (e) {
          console.log(e);
          yield put(stopSpinner());
        }
      } else {
        yield put(stopSpinner());
        yield put(
          requestShowNotification({
            message: "Biometrics not Scanned for the Job",
            type: "is-warning"
          })
        );
        // console.log("Please Add Fingerpints to goto tenprint Verify");
      }
    }
  } else {
    let ActionTpe = yield select(ActionSelector);

    if (
      ActionTpe !== "Custom Search Criminal" &&
      ActionTpe !== "Custom Search Civil" &&
      ActionTpe !== "Custom Search Latent"
    ) {
      const { payload: historydata } = action;

      const res = yield call(Api, {
        url: url.action.actionResult,
        method: "POST",
        isJwtRequired: true,
        data: historydata.JOBID
      });
      if (res) {
        if (res.note !== undefined && res.note !== ""&& res.note !== null) {
          yield put(requestStoreNote(res.note));
          yield put(stopSpinner());
          yield put(requestShowModal({ modalType: "NOTE_SHOW_MODAL" }));
        } else {
          yield put(stopSpinner());
        }
      }
    } else {
      yield put(stopSpinner());
      let Personinfos = {};
      const { payload: historydata } = action;
      const res = yield call(Api, {
        url: url.action.actionResult,
        method: "POST",
        isJwtRequired: true,
        data: historydata.JOBID
      });
      if (res) {
        let TenprintData = {
          leftfingers: {},
          rightfingers: {},
          leftmatches: {},
          rightmatches: {},
          personinfos: res.matchPersons,
          ProbeMugshotInfo: {},
          ProbePersonInfo: {},
          database: res.database,
        };
        yield put(requestSetTenprintVerify(TenprintData));
        yield put(setPersonData(res.matchPersons[0]));
        yield put(setCustomSearchDatabase(res.database));
        yield put(setCustomSearchResponse(res));
        const jobid = yield select(selectedJobSelector);
        const searchTextRes = yield call(Api, {
          url: url.job.getSearchTexts,
          method: "POST",
          isJwtRequired: true,
          data: jobid
        });
        if (searchTextRes) {
          yield put(requestSetCustomSearchID(searchTextRes));
        }
      }

      yield put(push("/authenticated/customsearch/searchverify"));
    }
  }
}

export function* callrequestUnMatch(){
  return true;
}

export function* callRequestMatchedPersonChanged(action) {
  const { payload: MatchedData } = action;
  let Decision = "";

  let MatchedPersonInfo = yield select(MatchedPersonInfoSelector);

  if (MatchedPersonInfo.id !== "" && MatchedPersonInfo.id !== undefined) {
    // let Send_data = {
    //   matchedPersonId: MatchedPersonInfo.id,
    //   decision: "No Matched"
    // };
    // const res = yield call(Api, {
    //   url: url.action.setMatchedPersonDecision,
    //   method: "PUT",
    //   isJwtRequired: true,
    //   data: Send_data
    // });
  }

  if (MatchedData.Selected) {
    Decision = "Matched";
  } else {
    Decision = "No Matched";
  }
  let Send_data = {
    matchedPersonId: MatchedData.userID,
    decision: Decision
  };

  // const res = yield call(Api, {
  //   url: url.action.setMatchedPersonDecision,
  //   method: "PUT",
  //   isJwtRequired: true,
  //   data: Send_data
  // });


  yield put(recivedMatchedPersonChanged(MatchedData));

  // if (res.message) {
  //   let output = {...MatchedData};
  //   if ( output.PersonInfo ) {
  //     // output.PersonInfo.decision = output.PersonInfo.decision === "No Matched" ? "Matched" : "No Matched";
  //     output.PersonInfo.decision = Decision;
  //   }
  //   yield put(recivedMatchedPersonChanged(output));
  // }

}
export function* callRequestModalData(action) {
  const ActionID = yield select(selectedActionSelector);

  let Send_data = {
    actionId: ActionID
  };
  const res = yield call(Api, {
    url: url.action.getMatchSubsequenceActions,
    method: "POST",
    isJwtRequired: true,
    data: Send_data
  });
  if (res) {
    yield put(recivedModalData(res));
    // ActionSelector
    yield put(resetActionType());
    yield put(requestShowModal({ modalType: "TENPRINT_VERIFY_CONFIRM" }));
    if (yield select(ActionSelector) === "") {
    } else {
      // yield put(push("/authenticated/jobqueue"))
    }
  }
}
export function* callRequestShowAdjudicator(action) {
  const { payload: OpenAdjudicator } = action;

  let selectedRow = yield select(SelectedRowSelector);
  let Send_data = {
    probeBiometricsId: OpenAdjudicator.imageId
  };

  if (
    OpenAdjudicator.match &&
    !Array.isArray(OpenAdjudicator.match) &&
    OpenAdjudicator.match.matchedFingerprintID
  ) {
    Send_data.matchedBiometricsId = OpenAdjudicator.match.matchedFingerprintID;
  } else if (
    Array.isArray(OpenAdjudicator.match) &&
    OpenAdjudicator.match.length > 0
  ) {
    Send_data.matchedBiometricsId =
      OpenAdjudicator.match[0].matchedFingerprintID;
  }

  const res = yield call(Api, {
    url: url.action.getMatchedBiometricsImage,
    method: "POST",
    isJwtRequired: true,
    data: Send_data
  });

  if (res) {

    if ( res.matched ) {
      let finalData = {
        Result: res,
        status: OpenAdjudicator.status,
        Data: OpenAdjudicator
      };
      yield put(requestShowAdjudicatorSuccess(finalData));
    } else {
      yield put(
        requestShowNotification({
          message:translate("noMatchReturned") ,
          type: "is-warning"
        })
      );
      yield put( requestTogglePaneVisibility({ pane: "all", visible: true }) );
    }
  }
}
