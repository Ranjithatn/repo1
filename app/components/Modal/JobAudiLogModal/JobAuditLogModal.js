import React, { Fragment } from "react";
import { connect } from "react-redux";

import Modal from "../Modal";
import Title from "../../Title/Title";
import Button from "../../Button/Button";
import { requestHideModal } from "../../../actions/modal";
import { jobAuditLogDescriptionSelector } from "../../../selectors/auditlog";
import "./JobAuditLogModal.scss";

export const JobAuditLogModal = ({
    jobAuditLogDescription,
    formatMessage, 
    history,
    requestHideModal
}) => {
  return (
    <Modal
      title={formatMessage({ id: "Log Details" })}
      className="newAction-modal"
      requestHideModal={requestHideModal}
      content={
        <Fragment>
          <Title
            is="5"
            text={formatMessage({ id: "desc" })}
          />
          <span>
            {jobAuditLogDescription}
            </span>
          
        </Fragment>
      }
      buttons={
        <Fragment>
          <Button
            id="newAction-btn"
            className="is-primary"
            text={formatMessage({ id: "close" })}
            onClick={e => requestHideModal()}
          />
        </Fragment>
      }
      
    />
  );
};

const mapState = state => ({
  jobAuditLogDescription: jobAuditLogDescriptionSelector(state),
});
export default connect(mapState, {
  requestHideModal
})(JobAuditLogModal);
