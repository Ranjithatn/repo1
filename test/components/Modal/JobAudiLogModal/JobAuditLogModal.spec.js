import React from "react";
import toJson from "enzyme-to-json";
import { JobAuditLogModal } from "../../../../app/components/Modal/JobAudiLogModal/JobAuditLogModal";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";
import { databaseSelector } from "../../../../app/selectors/tenprint";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id
}

describe('<JobAuditLogModal />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <JobAuditLogModal {...props} />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
