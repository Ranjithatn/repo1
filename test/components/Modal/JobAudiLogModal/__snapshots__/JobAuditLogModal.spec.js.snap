// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobAuditLogModal /> should render without crashing 1`] = `
<Component
  formatMessage={[Function]}
>
  <Modal
    buttons={
      <UNDEFINED>
        <Button
          className="is-primary"
          id="newAction-btn"
          onClick={[Function]}
          text="close"
        />
      </UNDEFINED>
    }
    className="newAction-modal"
    content={
      <UNDEFINED>
        <Title
          as="h1"
          is="5"
          text="desc"
        />
        <span />
      </UNDEFINED>
    }
    title="Log Details"
  >
    <div
      className="modal is-active newAction-modal"
      style={
        Object {
          "zIndex": "50",
        }
      }
    >
      <div
        className="modal-background"
      />
      <div
        className="modal-card"
      >
        <header
          className="modal-card-head"
        >
          <p
            className="modal-card-title"
          >
            Log Details
          </p>
          <button
            aria-label="close"
            className="delete is-large"
            name="btn-modal-close"
          />
        </header>
        <section
          className="modal-card-body"
        >
          <Title
            as="h1"
            is="5"
            text="desc"
          >
            <h1
              className="title is-5"
            >
              desc
            </h1>
          </Title>
          <span />
        </section>
        <footer
          className="modal-card-foot"
        >
          <Button
            className="is-primary"
            id="newAction-btn"
            onClick={[Function]}
            text="close"
          >
            <button
              className="button is-primary"
              disabled={false}
              id="newAction-btn"
              onClick={[Function]}
              style={Object {}}
              type="button"
            >
              <span>
                close
              </span>
            </button>
          </Button>
        </footer>
      </div>
    </div>
  </Modal>
</Component>
`;
